src/main/telemetry/smartport.c

```diff
@@ -161,8 +160,8 @@ static void smartPortDataReceive(uint16_t c)
     static uint8_t lastChar;
     if (lastChar == FSSP_START_STOP) {
         smartPortState = SPSTATE_WORKING;
-        smartPortLastRequestTime = now;
-        if (c == FSSP_SENSOR_ID1) {
+        if (c == FSSP_SENSOR_ID1 && (serialTotalBytesWaiting(smartPortSerialPort) == 0)) {
```

Comment:
`serialTotalBytesWaiting` wont help here - when using callback, input buffer is skipped completely ...